packages/contracts/utils/stateChannel.ts

```diff
@@ -0,0 +1,191 @@
+import * as ethers from "ethers";
+import { HIGH_GAS_LIMIT, signMessageBytes, ZERO_ADDRESS } from "../../test-utils/src/utils";
+import * as artifacts from "./buildArtifacts";
+import { Contract } from "./contract";
+import { Multisig } from "./multisig";
+import { encodeStruct, structDefinitionToEncoding } from "./structEncoding";
+
+const { keccak256 } = ethers.utils;
+
+export enum AssetType {
+  ETH,
+  ERC20
+}
+
+export interface TransferTerms {
+  assetType: AssetType;
+  limit: ethers.utils.BigNumber;
+  token?: string;
+}
+
+export interface App {
```

Comment:
We should really start sharing more code between the machine and this. See https://github.com/counterfactual/machine/blob/master/src/middleware/cf-operation/types.ts#L39

Reply:
I also think using explicit types instead of `string` everywhere helps readability, a lot, especially for new readers.

Reply:
I agree and I agree. This is why @cf19drofxots' work on linking packages is so important.